contracts/btc-finality/src/contract.rs

```diff
@@ -290,14 +285,18 @@ fn send_rewards_msg(
             })
         })
         .collect::<StdResult<Vec<_>>>()?;
+
+    // Calculate total rewards from individual rewards
+    let total_rewards: u128 = fp_rewards.iter().map(|r| r.reward.u128()).sum();
```

Comment:
Is it possible to 

- let `distribute_rewards_in_range` return the `total_rewards`, and
- pass `total_rewards` as a parameter in the `send_rewards_msg` function?

Summing is `O(n)` operation that ideally we can avoid

Reply:
Don't think it's possible as in `distribute_rewards_in_range` we also need to iterate the db to do the same thing. One way though, is to revert [c5fb579](https://github.com/babylonlabs-io/cosmos-bsn-contracts/pull/291/commits/c5fb5790d7298dc54821056606641fad9c4f2295) to re-introduce `TOTAL_PENDING_REWARDS`

Reply:
Actually I think it's better not to re-introduce `TOTAL_PENDING_REWARDS` as we need to maintain an extra db and should keep its value consistent with the fp rewards sum.

The summing complexity is acceptable as the num of active fps are limited by the params